go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -180,44 +194,71 @@ func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddres
 			action = "drop"
 		}
 
-		ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
-		if err != nil {
-			return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
-		}
-		if !utilnet.IsIPv6(ipAddress) {
-			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+		if rule.to.cidrSelector != "" {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if !utilnet.IsIPv6(ipAddress) {
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+			} else {
+				match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			}
+
+			if len(rule.ports) > 0 {
+				match = fmt.Sprintf("%s && ( %s )\"", match[:len(match)-1], egressGetL4Match(rule.ports))
+			}
+			for _, joinSwitch := range joinSwitches {
+				err = createACLRule(defaultStartPriority-rule.id, match, action, ef.namespace, joinSwitch)
+				if err != nil {
+					return err
+				}
+			}
 		} else {
-			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			// rule based on DNS NAME
+			var portMatch string
+			if oc.egressFirewallDNS == nil {
+				var err error
+				oc.egressFirewallDNS, err = NewEgressDNS()
```

Comment:
does NewEgressDNS not return any error? I don't really see the point of initializing it to zero value beforehand.

Reply:
Because `oc.egressFirewallDNS` is not a variable (I am pretty sure as a struct field it is a primary expression) it cannot be on the left hand side of the shorthand `:=` which is short for variable declaration, this requires that `err` be declared beforehand.

Reply:
He's asking: why bother having a constructor with no parameters?

Reply:
Agreed - get rid of this, just always set oc.egressFirewallDNS on initialization. No need to lazy-initialize.

Reply:
moved the initialization to inside the CRD watcher